### DISCUSSION TITLE
General observation about Debug vs Release builds

### DISCUSSION THREAD

**Discussion by udif** [MikePopoloski/slang#1061] [General]
I ran a large `slang` run on a full SoC. At first, I've used a Debug build I had already compiled, and it took close to an hour and ~70GB RAM to finish.
I later recompiled a Release build, and it finished the same run in less than 3 minutes... (it was quick enough I haven't noticed how much RAM it used).
I wonder why there is such a huge performance loss for debug builds. I was expecting somewhere between 10% to 50% speed loss.

**Comment by MikePopoloski** [comment]
I don't have enough information to say; it probably varies drastically based on your design, the compiler you used to build slang, various options and settings in your environment, etc. Maybe your debug builds have enabled debug heaps or leak checkers or whatever. Maybe there's some debugging checks in slang that have O(N^2) behavior somewhere so that it's only noticeable in large designs in debug builds. If you want to do some debugging and tweaks for performance I'm happy to accept them but I don't really have any place to start otherwise.

**Comment by udif** [comment]
I was just using the default cmake flags for selecting debug builds. I haven't expected anything from it other than adding the `-g` flag for gcc.
My post was just an observation or a hint to other users rather than a request for help.
